### PR TITLE
Updated JDOX links so they don't produce 404 errors

### LINKED ISSUE #889: JDox links to Instrument documentation pages broken
The links to the JDOX documentation per instrument seems to be broken. 

From the main page, choosing 'Documentation' in the drop down from any instrument returns the following: 
Don't feel lost...

JDox content at the link you’re using may have been moved or deleted.
Please use the navigation sidebar or the search bar to find the information you’re seeking 

The links need to be checked and updated.

### PR DESCRIPTION
**Pull request by york-stsci**
In particular, this updates the instrument-specific "Documentation" links, as well as the JDox URL returned on an exception from utils.py filename_parser().

Closes #889

### CODE REVIEW

**Comment by york-stsci** [comment]
None of the PEP8 comments have to do with the code that I changed. 

@bourque they all seem like easy fixes, do you want me to just fix them even though they're not to do with this particular update?

**Comment by bourque** [comment]
@york-stsci That would be great, thanks!

**Comment by york-stsci** [comment]
@bourque I think this is ready for review-and-merge.

File: jwql/utils/constants.py [hunk 1]
```diff
@@ -345,12 +345,12 @@
...
+    TIME_SERIES_SUFFIX_TYPES + NIRCAM_CORONAGRAPHY_SUFFIX_TYPES + \
+    NIRISS_AMI_SUFFIX_TYPES + WFSC_SUFFIX_TYPES
 
 # Instrument Documentation Links
-URL_DICT = {'fgs': 'https://jwst-docs.stsci.edu/jwst-observatory-hardware/fine-guidance-sensor',
-            'miri': 'https://jwst-docs.stsci.edu/mid-infrared-instrument',
-            'niriss': 'https://jwst-docs.stsci.edu/near-infrared-imager-and-slitless-spectrograph',
-            'nirspec': 'https://jwst-docs.stsci.edu/near-infrared-spectrograph',
-            'nircam': 'https://jwst-docs.stsci.edu/near-infrared-camera'}
+URL_DICT = {'fgs': 'https://jwst-docs.stsci.edu/jwst-observatory-hardware/jwst-fine-guidance-sensor',
```
Comment (bourque) [inline jwql/utils/constants.py:352]: Mildly interesting that JDox updated their URLs to have `jwst-` in them.  I wonder why!

**Review by bourque — APPROVED** [review]
Thanks @york-stsci, looks great!

**Pull request merged**